samples/RhythmGame/src/main/cpp/audio/AAssetDataSource.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include <utils/logging.h>
+#include "AAssetDataSource.h"
+
+
+AAssetDataSource* AAssetDataSource::newFromAssetManager(AAssetManager *assetManager,
+														const char *filename,
+														const int32_t channelCount) {
+
+	// Load the backing track
+	AAsset* asset = AAssetManager_open(assetManager, filename, AASSET_MODE_BUFFER);
```

Comment:
This whole file seems to have tabs. Four spaces are preferred.

Reply:
That's weird. In both Android Studio and CLion I have my editor set to 4 spaces. Fixed now.